compiler/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -620,6 +621,47 @@ impl<'a> FunctionContext<'a> {
         Ok(Self::unit_value())
     }
 
+    /// Codegens a while loop, creating three new blocks in the process.
+    /// The return value of a while is always a unit literal.
+    ///
+    /// For example, the loop `while cond { body }` is codegen'd as:
+    ///
+    /// ```text
+    ///   jmp while_entry()
+    /// while_entry:
+    ///   v0 = ... codegen cond ...
+    ///   jmpif v0, then: while_body, else: while_end  
+    /// while_body():
+    ///   v3 = ... codegen body ...
+    ///   br while_entry()
+    /// while_end():
+    ///   ... This is the current insert point after codegen_for finishes ...
+    /// ```
+    fn codegen_while(&mut self, while_: &While) -> Result<Values, RuntimeError> {
```

Comment:
Would be interesting if we could desugar for-loops into this. I would need to think on whether it makes the code cleaner or more messy (with the need for an induction variable)

Reply:
I can try, but I'm not sure. I think the only "duplicated" code is `codegen_for` and `codgen_while`, but they are a bit different because of the loop var. And incrementing the loop var currently does an `add_unchecked` and I'm not sure we have an AST node to do that.